BUG: Tests fail on Windows due to encoding error

The CI tests fail on Windows runners, as shown in #3029

Example run: https://github.com/slundberg/shap/actions/runs/5356313208/jobs/9715685628?pr=3029

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\hostedtoolcache\windows\Python\3.11.4\x64\Lib\site-packages\shap\datasets.py:86: in imdb
    data = f.readlines()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <encodings.cp1252.IncrementalDecoder object at 0x0000027237A2C1D0>
input = b' the biggest crime of all: It doesn\'t finish! It simply ends as if it\'s a commercial break away from the rest of t...own a copy of Charlie\'s Angels (2000) - not a quality film, but it makes me laugh and allows me to unwind for a while'
final = False

    def decode(self, input, final=False):
>       return codecs.charmap_decode(input,self.errors,decoding_table)[0]
E       UnicodeDecodeError: 'charmap' codec can't decode byte 0x8d in position 3913: character maps to <undefined>

C:\hostedtoolcache\windows\Python\3.11.4\x64\Lib\encodings\cp1252.py:23: UnicodeDecodeError
```

--- a/shap/datasets.py
+++ b/shap/datasets.py
@@ -82,7 +82,7 @@ def imdb(display=False, n_points=None): # pylint: disable=unused-argument
     Paper to cite when using the data is: http://www.aclweb.org/anthology/P11-1015
     """
 
-    with open(cache(github_data_url + "imdb_train.txt")) as f:
+    with open(cache(github_data_url + "imdb_train.txt"), encoding="utf-8") as f:
         data = f.readlines()
     y = np.ones(25000, dtype=bool)
     y[:12500] = 0

